akka-stream-tests/src/test/scala/akka/stream/io/TlsSpec.scala

```diff
@@ -0,0 +1,312 @@
+package akka.stream.io
+
+import java.security.{ KeyStore, SecureRandom }
+import javax.net.ssl.{ TrustManagerFactory, KeyManagerFactory, SSLContext }
+import akka.stream.{ Graph, BidiShape, ActorFlowMaterializer }
+import akka.stream.scaladsl._
+import akka.stream.ssl._
+import akka.stream.testkit.{ TestUtils, AkkaSpec }
+import akka.util.ByteString
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.collection.immutable
+import scala.util.Random
+import akka.stream.stage.AsyncStage
+import akka.stream.stage.AsyncContext
+import java.util.concurrent.TimeoutException
+import akka.actor.ActorSystem
+import javax.net.ssl.SSLSession
+import akka.pattern.{ after ⇒ later }
+import scala.concurrent.Future
+import java.net.InetSocketAddress
+
+object TlsSpec {
+
+  val rnd = new Random
+
+  def initSslContext(): SSLContext = {
+
+    val password = "changeme"
+
+    val keyStore = KeyStore.getInstance(KeyStore.getDefaultType)
+    keyStore.load(getClass.getResourceAsStream("/keystore"), password.toCharArray)
+
+    val trustStore = KeyStore.getInstance(KeyStore.getDefaultType)
+    trustStore.load(getClass.getResourceAsStream("/truststore"), password.toCharArray)
+
+    val keyManagerFactory = KeyManagerFactory.getInstance(KeyManagerFactory.getDefaultAlgorithm)
+    keyManagerFactory.init(keyStore, password.toCharArray)
+
+    val trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
+    trustManagerFactory.init(trustStore)
+
+    val context = SSLContext.getInstance("TLS")
+    context.init(keyManagerFactory.getKeyManagers, trustManagerFactory.getTrustManagers, new SecureRandom)
+    context
+  }
+
+  class Timeout()(implicit system: ActorSystem) extends AsyncStage[ByteString, ByteString, Unit] {
```

Comment:
This is an overall timeout, not an idle-timeout right? Maybe it deserves a comment.